src/librustc_mir/borrow_check/nll/region_infer/mod.rs

```diff
@@ -494,20 +462,18 @@ impl<'tcx> RegionInferenceContext<'tcx> {
         while let Some(constraint_idx) = dirty_list.pop() {
             clean_bit_vec.insert(constraint_idx.index());
 
-            let constraint = &self.constraints[constraint_idx];
+            let constraint = &self.constraints.inner()[constraint_idx];
```

Comment:
this would be `outlives_constraints[constraint_idx]` with the `Index` impl I proposed